src/cpp/session/modules/SessionRAddins.R

```diff
@@ -14,3 +14,36 @@
 #
 
 assign(".rs.rAddinsEnv", new.env(parent = emptyenv()), envir = .rs.toolsEnv())
+
+.rs.addJsonRpcHandler("prepare_for_addin", function()
+{
+   assign(
+      ".rs.addins.savedShinyResponseFilter",
+      getOption("shiny.http.response.filter"),
+      envir = .rs.toolsEnv())
+   
+   options(shiny.http.response.filter = .rs.addins.shinyResponseFilter)
+})
+
+.rs.addFunction("addins.shinyResponseFilter", function(request, response)
+{
+   # bail if this isn't the root response
+   if (!identical(request$PATH_INFO, "/"))
+      return(response)
+   
+   # remove our old handler
+   options(shiny.http.response.filter = .rs.addins.savedShinyResponseFilter)
```

Comment:
This implies we'll only filter the very first request for `/`. Is it possible for `/` to be requested more than once? (e.g. can the app trigger a reload of the frame?)

Reply:
I think it's possible, but I'm not aware of any addins that do this. That said, re: previous comment if we instead clear the handler with some other signal that the associated addin has stopped running we would be clear (e.g. ConsolePromptEvent)